kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs

```diff
@@ -632,6 +632,41 @@ impl<'tcx> GotocCtx<'tcx> {
             )
             .with_is_extern(true)
         });
+        (func, funct)
+    }
+
+    /// For a given function instance, generates an expression for the function symbol of type Code.
+    pub fn codegen_func_expr(&mut self, instance: Instance<'tcx>, span: Option<&Span>) -> Expr {
+        let (func, funct) = self.ensure_func(instance);
         Expr::symbol_expression(func, funct).with_location(self.codegen_span_option(span.cloned()))
     }
+
+    /// For a given function instance, generates a zero-sized dummy symbol (because FnDefs are zero-sized).
+    pub fn codegen_func_expr_zst(&mut self, instance: Instance<'tcx>, span: Option<&Span>) -> Expr {
+        let (func, _funct) = self.ensure_func(instance);
+        let fn_struct_ty = self.ensure_fndef_zst(instance);
+        // This zero-sized object that a function name refers to in Rust is globally unique, so we create such a global object.
+        let fn_singleton_name = format!("{func}::FnDefSingleton");
+        let fn_singleton = self.ensure_global_var(
+            &fn_singleton_name,
+            false,
+            fn_struct_ty.clone(),
+            Location::none(),
+            |_, _| None, // zero-sized, so no initialization necessary
+        );
+        fn_singleton.with_location(self.codegen_span_option(span.cloned()))
```

Comment:
Why clone the `span`

Reply:
Because `codegen_span_option` expects an `Option<Span>`, not `Option<&Span>`.